property/index.js

```diff
@@ -115,54 +118,128 @@ module.exports = yeoman.generators.Base.extend({
       {
          name: 'defaultValue',
          message: 'Default value[leave blank for none]:',
-         default: null
+         default: null,
+         when: function(answers) {
+          return answers.type !== null && 
+            answers.type !== 'buffer'&&
+            answers.type !== 'any'&&
+            typeChoices.indexOf(answers.type) !== -1;
+        }
       }
     ];
+
     this.prompt(prompts, function(answers) {
+      debug('answers: %j', answers);
       this.name = answers.name || this.name;
       if (answers.type === 'array') {
         var itemType =  answers.customItemType || answers.itemType;
         this.type = itemType ? [itemType] : 'array';
       } else {
         this.type = answers.customType || answers.type;
       }
-      this.required = answers.required;
-      this.defaultValue = answers.defaultValue;
+
+      this.propDefinition = {
+        name: this.name,
+        type: this.type
+      };
+      if (answers.required) {
+        this.propDefinition.required = true;
+      }
+
+      if (answers.defaultValue) {
+        this.propDefinition.type = answers.type;
```

Comment:
Actually, your implementation is missing the "back" part.

```
$ yo loopback:property
? Select the model: Avatar
? Enter the property name: nums
? Property type: array
? The type of array items: number
? Required? No
? Default value[leave blank for none]: 1,2,3
```

Here is the property definition created in `common/models/avatar.json`:

```
    "nums": {
      "type": "array",
      "itemType": "number",
      "default": [
        1,
        2,
        3
      ]
    }
```

As you see, there is `type` and `itemType` instead of `type: ['number']`.